IngressMaxFields.user.js

```diff
@@ -68,67 +70,89 @@ function wrapper() {
 			}
 		});
 		return c;
+	};
+	self.inBounds = function(portal) {
+		if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
+			return self.portalInDrawnItems(portal);
+		} else {
+			return self.portalInScreen(portal);
+		}
+	};
+	self.genStr = function genStr(p,x) {
+		var href = "https://www.ingress.com/intel?ll=" + p._latlng.lat + "," + p._latlng.lng + "&z=17&pll=" + p._latlng.lat + "," + p._latlng.lng;
+		var str = p.options.data.title;
+		str = str.replace(/\"/g, "\\\"");
+		str = str.replace(";"," ");
+		if (window.plugin.keys && (typeof window.portals[x] !== "undefined")) {
+			var keyCount = window.plugin.keys.keys[x] || 0;
+			str = str + ";" +href +";" + keyCount;
+		} else {
+			str = str + ";" + href;
+		}
+		return str;
 	}
 
-	self.gen = function gen() {
-			var o = [];
-			var antal = 0;
-			var tooMany = 0;
-
-			if (window.plugin.drawTools && window.plugin.drawTools.drawnItems.getLayers().length) {
-				var inBounds = function(portal) {
-					return self.portalInDrawnItems(portal);
-				}
-				var string = "Portal selection based on drawTools boundaries.";
-			} else {
-				var inBounds = function(portal) {
-					return self.portalInScreen(portal);
-				}
-				var string = "Portal selection based on screen boundaries.";
+	self.ManagePortals = function ManagePortals(obj,portal,x) {
+		if (obj.count <= 50)
+		{
+			if(self.inBounds(portal))
+			{
+				var str = self.genStr(portal,x);
+				obj.list.push(str)
+				obj.count = obj.count + 1;
 			}
 
-			for (var x in window.portals) {
-				var p = window.portals[x];
-				if (antal < 50) {
-					if (inBounds(p)) {
-						antal = antal + 1;
-						var href = 'https://www.ingress.com/intel?ll=' + p._latlng.lat + ',' + p._latlng.lng + '&z=17&pll=' + p._latlng.lat + ',' + p._latlng.lng;
-						var str1 = p.options.data.title.replace(/\"/g, "\\\"");
-						var str2 = str1.replace(';', ' ');
+		}
+		return obj;
 
-						//check for keys plugin
-						if (window.plugin.keys) {
-							var keyCount = window.plugin.keys.keys[x] || 0;
-							o.push(str2 + ";" + href + ";" + keyCount);
-						} else {
-							o.push(str2 + ";" + href);
-						}
-					}
-				} else {
-					if (tooMany == 0) {
-						tooMany = 1;
-					}
-				}
+	};
+	self.checkPortals= function checkPortals(portals) {
+		var obj = {
+			list:[],
+			tooMany:false,
+			count:0
+			};
+		for (var x in portals) {
+			if (typeof window.portals[x] !== "undefined") {
+				self.ManagePortals(obj,window.portals[x],x);
 			}
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ["count" is not defined.](https://www.codacy.com/app/stefan-nygren/IITC-Ingress-Maxfields-Exporter/file/1750482551/issues/source?bid=3039347&fileBranchId=3045939#l120)